cogeo_mosaic/model.py

```diff
@@ -24,3 +31,169 @@ class MosaicJSON(BaseModel):
     bounds: List[float] = Field([-180, -90, 180, 90])
     center: Optional[Tuple[float, float, int]]
     tiles: Dict[str, List[str]]
+
+    @classmethod
+    def from_dict(cls, mosaic_def: Dict):
+        return MosaicJSON(**dict(mosaic_def))
+
+    @classmethod
+    def from_features(
+        cls,
+        features,
+        minzoom: int,
+        maxzoom: int,
+        quadkey_zoom: Optional[int] = None,
+        **kwargs,
+    ):
+        MosaicJSON._create_mosaic(
+            features,
+            minzoom=minzoom,
+            maxzoom=maxzoom,
+            quadkey_zoom=quadkey_zoom,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_urls(
+        cls,
+        dataset_list: Sequence[str],
+        minzoom: int = None,
+        maxzoom: int = None,
+        max_threads: int = 20,
+        quiet: bool = True,
+        **kwargs,
+    ):
+        """Create mosaic definition from list of urls or files
+        """
+        if not quiet:
+            click.echo("Get files footprint", err=True)
+
+        features = get_footprints(dataset_list, max_threads=max_threads, quiet=quiet)
+
+        if minzoom is None:
+            minzoom = {feat["properties"]["minzoom"] for feat in features}
+            if len(minzoom) > 1:
+                warnings.warn(
+                    "Multiple MinZoom, Assets different minzoom values", UserWarning
+                )
+
+            minzoom = max(minzoom)
+
+        if maxzoom is None:
+            maxzoom = {feat["properties"]["maxzoom"] for feat in features}
+            if len(maxzoom) > 1:
+                warnings.warn(
+                    "Multiple MaxZoom, Assets have multiple resolution values",
+                    UserWarning,
+                )
+
+            maxzoom = max(maxzoom)
+
+        datatype = {feat["properties"]["datatype"] for feat in features}
+        if len(datatype) > 1:
+            raise Exception("Dataset should have the same data type")
+
+        return MosaicJSON.from_features(
+            features, minzoom=minzoom, maxzoom=maxzoom, quiet=quiet, **kwargs
+        )
+
+    @classmethod
+    def _create_mosaic(
```

Comment:
I think this could be `from_feature` directly.

Reply:
I feel that it's better to split it up for composability. It's a nice logical separation